catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -23,94 +25,104 @@
 import java.util.Set;
-import java.util.function.Supplier;
 import java.util.stream.Collectors;
 import org.apache.commons.lang.StringUtils;
 import org.codice.alliance.catalog.core.api.impl.types.SecurityAttributes;
 import org.codice.alliance.catalog.core.api.types.Security;
-import org.opensaml.core.xml.schema.XSString;
-import org.opensaml.saml.saml2.core.AttributeStatement;
+import org.codice.ddf.security.SystemHighAttributes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+/**
+ * This plugin sets security attributes on {@link Metacard}s to system high attribute values if none
+ * of the policies were able to apply security markings to this {@link Metacard}. The configuration
+ * for this plugin maps 6 security {@link Attribute} names to system high attribute names. This
+ * plugin only retrieves the system high attribute values when the configuration is updated.
+ */
 public class DefaultSecurityAttributeValuesPlugin implements PreIngestPlugin {
 
-  private final SecurityAttributes securityAttributes;
-
-  private final Supplier<Subject> subjectSupplier;
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(DefaultSecurityAttributeValuesPlugin.class);
 
-  private Map<String, Attribute> securityMarkings;
+  private static final String CLASSIFICATION_CONFIGURATION_KEY = "classification";
 
-  public static final String DEFAULTMARKINGS = "defaultMarkings";
+  private static final String RELEASABILITY_CONFIGURATION_KEY = "releasability";
 
-  static final String CLASSIFICATION_KEY = "classification";
+  private static final String CODEWORDS_CONFIGURATION_KEY = "codewords";
 
-  static final String CODEWORDS_KEY = "codewords";
+  private static final String DISSEMINATION_CONTROLS_CONFIGURATION_KEY = "disseminationControls";
 
-  static final String RELEASABILITY_KEY = "releasability";
+  private static final String OTHER_DISSEMINATION_CONTROLS_CONFIGURATION_KEY =
+      "otherDisseminationControls";
 
-  static final String DISSEMINATION_CONTROLS_KEY = "disseminationControls";
+  private static final String OWNER_PRODUCER_CONFIGURATION_KEY = "ownerProducer";
 
-  static final String OTHER_DISSEMINATION_CONTROLS_KEY = "otherDisseminationControls";
+  private static final String DEFAULT_MARKINGS_TAG = "defaultMarkings";
 
-  static final String OWNER_PRODUCER_KEY = "ownerProducer";
+  private final SecurityAttributes securityAttributes;
 
-  private static final Map<String, Set<String>> SYS_HIGH_TO_METACARD_ATTRIBUTE_MAPPING =
-      new HashMap<>();
+  private final SystemHighAttributes systemHighAttributes;
 
-  public DefaultSecurityAttributeValuesPlugin(
-      SecurityAttributes securityAttributes, Map<String, Object> initialValues) {
-    this(
-        securityAttributes,
-        initialValues,
-        () -> org.codice.ddf.security.common.Security.getInstance().getSystemSubject());
-  }
+  /**
+   * This map will contain an entry for all 6 of the metacard attribute names. It is {@code
```

Comment:
✏️  We say _6_ metacard attributes here and in the class description, but really the number depends on what's being passed in the `initialValues` map. We may want to update the comments and make that clearer.

Reply:
The `metacardAttributeNameToSystemHighAttributeNameMap` will _always_ contain 6 values (once it's initialized). If `initialValues` does not contain all 6 configuration values, an error will be thrown, and `metacardAttributeNameToSystemHighAttributeNameMap` will not updated.